services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/LinkOperationsBolt.java

```diff
@@ -112,6 +124,74 @@ public LinkOperationsBolt(Auth neoAuth) {
         return results;
     }
 
+    private List<FlowsResponse> getFlowIdsForLink(GetFlowIdsForLinkRequest request, Session session) {
+        SwitchId sourceSwitchId = request.getSource().getDatapath();
+        Integer sourcePort = request.getSource().getPortNumber();
+        SwitchId destinationSwitchId = request.getDestination().getDatapath();
+        Integer destinationPort = request.getDestination().getPortNumber();
+
+        ErrorMessage errorMessage = null;
+        if (sourceSwitchId == null || sourcePort == null || destinationSwitchId == null || destinationPort == null) {
+            logger.error("Not all parameters are specified.");
+            ErrorData errorData = new ErrorData(ErrorType.PARAMETERS_INVALID,
+                    "Not all parameters are specified.",
+                    "ISL does not exist.");
+            errorMessage = new ErrorMessage(errorData,
```

Comment:
Instead you need to throw an exception from the service method and a bolt will convert it to the corresponding error message.

Reply:
Changed this to throw the error.